tests/e2e_logging.py

```diff
@@ -1467,6 +1467,54 @@ def test_rekey(network, args):
     return network
 
 
+@reqs.description("Test UDP echo endpoint")
+@reqs.at_least_n_nodes(1)
+def test_udp_echo(network, args):
+    primary, _ = network.find_primary()
+    udp_interface = primary.host.rpc_interfaces["secondary_rpc_interface"]
+    host = udp_interface.public_host
+    port = udp_interface.public_port
+    LOG.info(f"Testing UDP echo server at {host}:{port}")
+
+    server_address = (host, port)
+    buffer_size = 1024
+    test_string = b"Some random text"
+
+    with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as s:
+        s.sendto(test_string, server_address)
+        recv = s.recvfrom(buffer_size)
+
+    text = recv[0]
+    LOG.info(f"Testing UDP echo server received {text}")
+    assert text == test_string
+
+
+def run_udp_tests(args):
+    # Register secondary interface as an UDP socket on first node
+    primary = args.nodes[0].rpc_interfaces["primary_rpc_interface"]
+    udp_interface = infra.interfaces.make_secondary_interface()
+    for interface in udp_interface.values():
+        interface.host = primary.host
+        interface.port = primary.port
+        interface.public_host = primary.public_host
+        interface.public_port = primary.public_port
+        interface.transport = "udp"
```

Comment:
Is there a strong reason to have the UDP interface listen on the same host:port as the main TCP interface? If not, I think it'd be simpler to do:
```python
primary.rpc_interfaces.update(infra.interfaces.make_secondary_interface(protocol="udp"))
```

Reply:
There was a reason, but not anymore. These ports are all `0` anyway, so they're different ones at runtime.

I did the `make_secondary_interface` change you proposed earlier, but I realised I still wanted to control other things (like host), but with the new registration process, even that's not needed any more.

I'll try again with the simpler call.